server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -135,13 +140,16 @@ public InstanceId getInstanceID() {
    * Should only be called by the Tablet server
    */
   public synchronized void setupCrypto() throws CryptoService.CryptoException {
-    if (cryptoService != null) {
-      throw new CryptoService.CryptoException("Crypto Service " + cryptoService.getClass().getName()
+    if (cryptoFactory != null) {
+      throw new CryptoService.CryptoException("Crypto Factory " + cryptoFactory.getClass().getName()
           + " already exists and cannot be setup again");
     }
 
     AccumuloConfiguration acuConf = getConfiguration();
-    cryptoService = CryptoServiceFactory.newInstance(acuConf, ClassloaderType.ACCUMULO);
+    cryptoFactory = CryptoFactoryLoader.newInstance(acuConf, ACCUMULO);
+    var props = acuConf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_CRYPTO_PREFIX);
+    CryptoEnvironment env = new CryptoEnvironmentImpl(CryptoEnvironment.Scope.WAL, null, null);
+    walCryptoService = cryptoFactory.getService(env, props);
```

Comment:
@ctubbsii recently in #2658 you changed ServerContext to use `Supplier` to provide a lot of the resources. What do you think about the 2 CryptoServices? I thought I had created `setupCrypto()` to take care of this but I guess that is not in main yet. I am currently working on merging conflicts for this branch.

Reply:
I think i figured it out. I saw that you dropped the method in #2658.